src/pages/workspaces/workspace/analysis/useAnalysisFiles.ts

```diff
@@ -0,0 +1,136 @@
+  create: CreateAnalysisFn
+  deleteFile: (path: AbsolutePath) => Promise<void>
+  pendingCreate: LoadedState<true, unknown>
+  pendingDelete: LoadedState<true, unknown>
+}
+
+export const useAnalysisFiles = (): AnalysisFileStore => {
+  const signal = useCancellation()
+  const [loading, setLoading] = useState(false)
+  const workspace: WorkspaceWrapper = useStore(workspaceStore)
+  const [analyses, setAnalyses] = useState<AnalysisFile[]>([])
+  const [pendingCreate, setPendingCreate] = useLoadedData<true>()
+  const [pendingDelete, setPendingDelete] = useLoadedData<true>()
+
+  const refresh = withHandlers([
+    withErrorReporting('Error loading analysis files'),
+    Utils.withBusyState(setLoading)
+  ], async (): Promise<void> => {
+    const workspaceInfo = workspace.workspace
+    const existingAnalyses: AnalysisFile[] = isGoogleWorkspaceInfo(workspaceInfo) ?
+      await Ajax(signal).Buckets.listAnalyses(workspaceInfo.googleProject, workspaceInfo.bucketName) :
+      await Ajax(signal).AzureStorage.listNotebooks(workspaceInfo.workspaceId)
+    setAnalyses(existingAnalyses)
+  })
+
+  const create = async (fullAnalysisName: string, toolLabel: ToolLabel, contents: any): Promise<void> => {
+    await setPendingCreate(async () => {
+      const workspaceInfo = workspace.workspace
+      isGoogleWorkspaceInfo(workspaceInfo) ?
+        await Ajax().Buckets.analysis(workspaceInfo.googleProject, workspaceInfo.bucketName, fullAnalysisName, toolLabel).create(contents) :
+        await Ajax().AzureStorage.blob(workspaceInfo.workspaceId, fullAnalysisName).create(contents)
+      await refresh()
+      return true
+    })
+  }
+
+  const deleteFile = async (path: AbsolutePath): Promise<void> => {
+    await setPendingDelete(async () => {
+      const workspaceInfo = workspace.workspace
+      if (isGoogleWorkspaceInfo(workspaceInfo)) {
+        await Ajax().Buckets.analysis(
+          workspaceInfo.googleProject,
+          workspaceInfo.bucketName,
+          getFileName(path),
+          getToolLabelFromFileExtension(getExtension(path))
+        ).delete()
+      } else {
+        await Ajax(signal).AzureStorage.blob(workspaceInfo.workspaceId, getFileName(path)).delete()
+      }
+      await refresh()
+      return true
+    })
+  }
+
+  useEffect(() => {
+    refresh()
+  }, [workspace]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (pendingCreate.status === 'Error') {
+      reportError('Error creating Analysis file.', pendingCreate.error)
+    }
+  }, [pendingCreate])
+
+  useEffect(() => {
+    if (pendingDelete.status === 'Error') {
+      reportError('Error deleting Analysis file.', pendingDelete.error)
+    }
+  }, [pendingDelete])
+  return {
+    refreshFileStore: refresh,
+    create,
+    deleteFile,
+    loadedState: { status: loading ? 'Loading' : 'Ready', state: analyses },
+    pendingCreate,
+    pendingDelete
+  }
+}
+
+// This is mainly a test utility in its current form
```

Comment:
If this function is a test utility, is this module the most appropriate place for it? Could it be moved to another module for test utilities, like the test data fixtures?

Reply:
Its a bit hairy, since it depends on some utilities in this file for convenience. As such, it's a circular dependency to include it in a testData file, which in turn tests some utilities in this file. This seems like the simplest solution that doesn't introduce yet more files